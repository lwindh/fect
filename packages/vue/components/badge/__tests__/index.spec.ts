import { mount } from '@vue/test-utils'
import Badge from '..'

const Wrapper = {
  components: {
    [Badge.name]: Badge
  },
  data() {
    return {
      type: 'default',
      dot: false,
      size: 'medium'
    }
  },
  template: `<div class="container">
   <fe-badge :type="type" :dot="dot" :size="size" :type-data="type">Badge</fe-badge>
 </div>`
}

describe('Badge', () => {
  it('should be render as a element', () => {
    const badge = mount(Badge)
    expect(() => badge.unmount()).not.toThrow()
  })
  it('should be supprot noramlize props', async () => {
    const wrapper = mount(Wrapper)
    const el = wrapper.find('.fect-badge')
    await wrapper.setData({ type: 'success' })
    expect(el.attributes('type-data')).toBe('success')
    await wrapper.setData({ size: 'large' })
    expect(el.attributes('class')).toContain('fect-badge--large')
    await wrapper.setData({ dot: true })
    expect(wrapper.find('.fect-badge__dot')).toBeTruthy()
    expect(wrapper.find('.fect-badge__dot').text()).toBe('')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
