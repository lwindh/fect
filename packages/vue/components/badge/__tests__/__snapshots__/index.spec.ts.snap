// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should be supprot noramlize props 1`] = `<div class="container"><span type-data="success" class="fect-dot fect-badge" style="font-size: 16px;"><!----></span></div>`;
